sdk/src/signer.rs

```diff
@@ -177,7 +176,58 @@ pub trait AsyncSigner: Sync {
         }
         None
     }
-    #[cfg(target_arch = "wasm32")]
+
+    /// OCSP response for the signing cert if available
+    /// This is the only C2PA supported cert revocation method.
+    /// By pre-querying the value for a your signing cert the value can
+    /// be cached taking pressure off of the CA (recommended by C2PA spec)
+    async fn ocsp_val(&self) -> Option<Vec<u8>> {
+        None
+    }
+
+    /// If this returns true the sign function is responsible for for direct handling of the COSE structure.
+    ///
+    /// This is useful for cases where the signer needs to handle the COSE structure directly.
+    /// Not recommended for general use.
+    fn direct_cose_handling(&self) -> bool {
+        false
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+#[async_trait(?Send)]
+pub trait AsyncSigner {
```

Comment:
Read the "Changes in this pull request" part of the PR description for info on why I had to duplicate this.

Reply:
I'm not sure we need to have send in the other cases, but I suppose it all depends on how it is being used.